target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4019-meshd01.dts

```diff
@@ -0,0 +1,390 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+    model = "Xiaomi Mesh Router D01";
+    compatible = "xiaomi,meshd01", "qcom,ipq4019";
+
+    chosen {
+	bootargs-append = " root=/dev/ubiblock0_0 clk_ignore_unused";
+    };
+
+    aliases {
+
+	serial0 = &blsp1_uart1;
+    };
+
+
+    memory {
+	device_type = "memory";
+	reg = <0x80000000 0x10000000>;
+    };
+
+    reserved-memory {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	ranges;
+
+	rsvd1@87000000 {
+	    reg = <0x87000000 0x500000>;
+	    no-map;
+	};
+
+	wifi_dump@87500000 {
+	    reg = <0x87500000 0x600000>;
+	    no-map;
+	};
+
+	rsvd2@87B00000 {
+	    reg = <0x87b00000 0x500000>;
+	    no-map;
+	};
+    };
+
+    keys {
+	compatible = "gpio-keys";
+
+	button@0 {
+	    label = "reset";
+	    linux,code = <KEY_RESTART>;
+	    gpios = <&tlmm 39 GPIO_ACTIVE_LOW>;
+	};
+
+	button@1 {
+	    label = "wps";
+	    linux,code = <KEY_WPS_BUTTON>;
+	    gpios = <&tlmm 40 GPIO_ACTIVE_LOW>;
+	};
+    };
+    soc {
```

Comment:
Missing empty line before this node.

Please take care of similar formatting issues below, I won't mark them again ...